clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4394,7 +4394,10 @@ public ListTransactionsResult listTransactions(ListTransactionsOptions options)
     public FenceProducersResult fenceProducers(Collection<String> transactionalIds, FenceProducersOptions options) {
         AdminApiFuture.SimpleAdminApiFuture<CoordinatorKey, ProducerIdAndEpoch> future =
             FenceProducersHandler.newFuture(transactionalIds);
-        FenceProducersHandler handler = new FenceProducersHandler(logContext);
+        if (options.timeoutMs() == null) {
+            options.timeoutMs(defaultApiTimeoutMs);
```

Comment:
I think it's best if we don't mutate the options if the user passed it in. If someone were to log the options somewhere else, they would see our injected default instead of the fact that no value was provided.

Reply:
Thanks for the good comment.

The main reason i mutated the options was to pass the same timeout value in both the `FenceProducersHandler` and in the `invokeDriver`.

I just noticed that this is already true as the [calcDeadlineMs](https://github.com/apache/kafka/blob/8d1a1dd7f2655e411e48ef5043b1385fd9804991/clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java#L469-L473) that is used by the `invokeDriver` is already returning the `defaultApiTimeoutMs` in the case of null timeout.